packages/flutter_tools/README.md

```diff
@@ -24,14 +27,33 @@ $ ../../bin/cache/dart-sdk/bin/dart bin/flutter_tools.dart
 followed by command line arguments, as usual.
 
 
-### Analyzing
+### Running the analyzer
 
 To run the analyzer on Flutter Tools, in this directory run:
 ```shell
 $ ../../bin/flutter analyze
 ```
 
-### Testing
+### Writing tests
+
+As with other parts of the Flutter repository, all changes in behavior [must be
+tested](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#write-test-find-bug).
+Tests live under the `test/` subdirectory.
+- Hermetic unit tests of tool internals go under `test/general.shard`.
+- Tests of tool commands go under `test/commands.shard`. Hermetic tests go under
+  its `hermetic/` subdirectory. Non-hermetic tests go under its `permeable`
+  sub-directory.
+- Integration tests (e.g. tests that run the tool in a subprocess) go under
+  `test/integration.shard`.
+
+In general, the tests for the code in a file called `file.dart` should go in a
+file called `file_test.dart` in the subdirectory that matches the behavior of
+the test.
+
+We measure [test coverage](https://codecov.io/gh/flutter/flutter) post-submit.
+A change should not regress test coverage.
```

Comment:
Might be too strong, since a change which purely deletes code could reduce overall coverage - if that code was more tested than average. Perhaps we should say that the diff of the coverage should not be too far off of the average coverage?

Reply:
Ah, right. Good call.

Reply:
I mean "coverage of diff" not "diff of coverage"